Makefile

```diff
@@ -127,15 +127,15 @@ MQTTLIB_A_TARGET = ${blddir}/lib${MQTTLIB_A}.so.${VERSION}
 MQTTLIB_AS_TARGET = ${blddir}/lib${MQTTLIB_AS}.so.${VERSION}
 MQTTVERSION_TARGET = ${blddir}/MQTTVersion
 
-CCFLAGS_SO = -g -fPIC $(CFLAGS) -Os -Wall -fvisibility=hidden -I$(blddir_work)
-FLAGS_EXE = $(LDFLAGS) -I ${srcdir} -lpthread -L ${blddir}
-FLAGS_EXES = $(LDFLAGS) -I ${srcdir} ${START_GROUP} -lpthread -lssl -lcrypto ${END_GROUP} -L ${blddir}
+CCFLAGS_SO = -g -fPIC $(CFLAGS) -D_GNU_SOURCE -Os -Wall -fvisibility=hidden -I$(blddir_work)
```

Comment:
Once you removed the _GNU_SOURCE from the source files, I guess that  _GNU_SOURCE should be in CMakeLists.txt too. Or maybe in a common file, so both build systems can use. Otherwise the CMake build will differ from the build performed by Make.

Reply:
In addition to adding the _GNU_SOURCE  compile definition to CMakeLists.txt, I believe that `anl` (asynchronous name lookup) must also be added to the set of LIBS_SYSTEM libraries.